prow/cmd/sub/main.go

```diff
@@ -93,16 +96,22 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 
-	var tokenGenerator func() []byte
+	var tokens []string
 	if flagOptions.pushSecretFile != "" {
-		var tokens []string
 		tokens = append(tokens, flagOptions.pushSecretFile)
+	}
+	if flagOptions.github.TokenPath != "" {
+		tokens = append(tokens, flagOptions.github.TokenPath)
+	}
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start(tokens); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+	tokenGenerator := secretAgent.GetTokenGenerator(flagOptions.pushSecretFile)
 
-		secretAgent := &secret.Agent{}
-		if err := secretAgent.Start(tokens); err != nil {
-			logrus.WithError(err).Fatal("Error starting secrets agent.")
-		}
-		tokenGenerator = secretAgent.GetTokenGenerator(flagOptions.pushSecretFile)
+	gitClient, err := flagOptions.github.GitClient(secretAgent, flagOptions.dryRun)
```

Comment:
Does this couple us to GitHub or does it work with Gerrit as well? We just need a git client so we should probably avoid coupling to GH.

Reply:
/hold

Reply:
This client is for inrepoconfig, which is not supported in gerrit yet. We can change here once inrepoconfig is supported in gerrit

Reply:
That makes sense, but I'm more wondering how this behaves without GitHub credentials supplied. In subscriber.go@198 we try to use the git client independent of whether we are using GitHub or Gerrit. It seems strange to use a client for a different code review provider when know it won't work and that also seems like it will result in misleading error messages.

 --------
Looks like we never try to use the git client [until after we've checked if InRepoConfig is enabled for the repo](https://github.com/kubernetes/test-infra/blob/a739353cae1e6c87cf22645918e4a4954acfdaf7/prow/config/config.go#L328-L330) and flag validation should pass if no GitHub flags are supplied so I think this is ok.
/shrug
/hold cancel